### PR TITLE
Fixed lost thin pool devicemapper docs

### LINKED ISSUE #22817: device-mapper-driver documentation update looses thinpool configuration info
Referring to #22166

The [last commit of the device-mapper documentation](https://github.com/docker/docker/commit/a7b2f87b0637a3711e90c0f1d0a99dd5d32bd60f#diff-52525eec69924c13d098f5d19b2d79adL254) by @chenchun clarified the overall use.
But as I go through the steps, I'm missing the part where the thinpool configuration on /data and /metadata is done.

The [diff shows that the previous version ](https://github.com/docker/docker/pull/22662/commits/3cf8c53515ff8144a63c6d4418738b5d4f4416a7)had this covered in lines 254 and following

### PR DESCRIPTION
**Pull request by mbentley**
**\- What I did**
Fixes #22817 

**\- How I did it**
Re-wrote docs missing and applied additional changes made since the docs were lost

**\- How to verify it**
Read the docs :)

**\- Description for the changelog**
First commit includes just what should be cherry picked into the `docs` branch; combination should go into `master`

Signed-off-by: Matt Bentley matt.bentley@docker.com

### CODE REVIEW

File: docs/userguide/storagedriver/device-mapper-driver.md [hunk 1]
```diff
@@ -221,106 +221,139 @@ assumes that the Docker daemon is in the `stopped` state.
...
+		 "storage-opts": [
+			 "dm.thinpooldev=/dev/mapper/docker-thinpool",
+			 "dm.use_deferred_removal=true"
+		 ]
+	 }
+	```
+15. Start the Docker daemon.
+
+	```bash
+	$ systemctl start docker
```
Comment (thaJeztah) [inline docs/userguide/storagedriver/device-mapper-driver.md]: Perhaps we should add a `systemctl daemon-reload` here to be sure changes are loaded (if they used a drop-in file, or modified the docker.service)

File: docs/userguide/storagedriver/device-mapper-driver.md [hunk 2]
```diff
@@ -221,106 +221,139 @@ assumes that the Docker daemon is in the `stopped` state.
...
+
+```bash
+$ journalctl -fu dm-event.service
+```
+
+If you run into repeated problems with thin pool, you can use the
+`dm.min_free_space` option to tune the Engine behavior. This value ensures that
+operations fail with a warning when the free space is at or near the minimum.
+For information, see <a
+href="https://docs.docker.com/engine/reference/commandline/dockerd/#storage-driver-options"
```
Comment (thaJeztah) [inline docs/userguide/storagedriver/device-mapper-driver.md]: Can you change this to a relative link;

```
../../../reference/commandline/daemon/#storage-driver-options
```
(for the first commit)

```
../../../reference/commandline/dockerd/#storage-driver-options
```

(for the second)

**Comment by thaJeztah** [comment]
While we're add it, could you download the image used at line 378
https://github.com/docker/docker/pull/22968/files#diff-52525eec69924c13d098f5d19b2d79adR378

```Markdown
![](http://farm1.staticflickr.com/703/22116692899_0471e5e160_b.jpg)
```

And add it to the https://github.com/docker/docker/tree/master/docs/userguide/storagedriver/images directory? (giving it a sensible name)

File: docs/userguide/storagedriver/device-mapper-driver.md [hunk 3]
```diff
@@ -412,10 +445,12 @@ The `Data Space` values show that the pool is 100GB total. This example extends
 1. List the sizes of the devices.
 
 	```bash
-	$ sudo ls -lh /var/lib/docker/devicemapper/devicemapper/
+	$ sudo ls -al /var/lib/docker/devicemapper/devicemapper/
 	total 1175492
-	-rw------- 1 root root 100G Mar 30 05:22 data
-	-rw------- 1 root root 2.0G Mar 31 11:17 metadata
+	drwx------ 2 root root         4096 Mar 29 02:45 .
```
Comment (thaJeztah) [inline docs/userguide/storagedriver/device-mapper-driver.md]: could you change this back? the example below also uses `lh` (probably easier to read)

**Comment by thaJeztah** [comment]
Thanks so much for doing this @mbentley! Couple of nits, but looking good

ping @vdemeester  - for reference; I asked Matt to use two separate commits, and keep the `docker daemon -> dockerd` rename separate, so that we can cherry-pick into the docs branch

**Comment by mbentley** [comment]
Thanks for the nits @thaJeztah.  Updated the commits.  Let me know if I missed anything.

**Comment by vdemeester** [comment]
LGTM 🐼
Ok for the commits :angel:

**Comment by thaJeztah** [comment]
awesome, thanks @mbentley!

**Pull request merged**